weasyl/searchtag.py

```diff
@@ -167,7 +167,7 @@ def is_tag_restriction_pattern_valid(text):
     return False
 
 
-def associate(userid, tags, submitid=None, charid=None, journalid=None):
+def associate(userid, tags, submitid=None, charid=None, journalid=None, artistid=None, optoutid=None):
```

Comment:
This method is becoming monstrous and the way things are being passed into it is imo an anti-pattern. `get_targetid` is also bad.

`artistid` is a bad name because it doesn't suggest what's it for. Consider `preferred_tags_userid` and `optout_tags_userid`.